spec/units/listing_spec.rb

```diff
@@ -1,19 +1,33 @@
 require 'listing.rb'
 
 describe Listing do
-  
+
   describe 'self.create' do
     it 'adds a listing to the database' do
-      pg_instance_returned = Listing.create('testtitle')
-      expect(pg_instance_returned[0]['title']).to eq 'testtitle'
+      conn = PG.connect(dbname: 'abodenb_test')
+      Listing.create(title: 'testtitle', description: 'test description', price: '100')
+      result = conn.exec("SELECT title FROM listings;")
+      expect(result[0]['title']).to eq 'testtitle'
     end
   end
 
   describe 'self.all' do
-    it 'returns an array of all listings in the database' do
+    it 'returns an array of listing objects of all listings in the database' do
       conn = PG.connect(dbname: 'abodenb_test')
       conn.exec("INSERT INTO listings (title) VALUES ('testtitle');")
-      expect(Listing.all).to eq ['testtitle']
+      conn.exec("INSERT INTO listings (title) VALUES ('testtitle2');")
+      expect(Listing.all.first).to be_kind_of(Listing)
+      expect(Listing.all.last).to be_kind_of(Listing)
+      expect(Listing.all.length).to eq 2
+      expect(Listing.all.first.title).to eq 'testtitle'
+      expect(Listing.all.last.title).to eq 'testtitle2'
+    end
+  end
+
+  describe '#title' do
+    it 'returns the title of the listing' do
+      pg_instance_returned = Listing.create(title: 'testtitle', description: 'test description', price: '100')
```

Comment:
should price be passed as an integer or float rather than a string?

Reply:
Good point - this also made us add a test for price, and test that it got returned as an integer rather than a string. SQL by default returns everything as a string so updated our code to pass the new test too.